client/reducers/articlesReducer.js

```diff
@@ -0,0 +1,41 @@
+import { data as sd } from 'sharify'
+import { clone } from 'lodash'
+import { actions } from 'client/actions/articlesActions'
+import { actions as editActions } from 'client/actions/editActions'
+
+export const initialState = {
+  articles: sd.ARTICLES,
+  articlesInSession: sd.ARTICLES_IN_SESSION || {}
+}
+
+export function articlesReducer (state = initialState, action) {
+  switch (action.type) {
+    case actions.EDITED_ARTICLES_RECEIVED: {
+      return {
+        ...state,
+        articlesInSession: action.payload
+      }
```

Comment:
I believe there's a lib called [updeep](https://github.com/substantial/updeep) being used elsewhere; what it does is allow for easy / fast / pseudo-immutable updates to nested data:

```javascript
import u from 'updeep'
...
case actions.EDITED_ARTICLES_RECEIVED: {
  return u({
    articlesInSession: action.payload,
    ...
    foo: {
      baz: [
        { ... }
      ]
    }
  }, state)
}

Reply:
Yes, I've seen in it in `editReducer.js` I went ahead and used it in this reducer module as well